watchtower/wtclient/backup_task.go

```diff
@@ -192,13 +192,25 @@ func (t *backupTask) bindSession(session *wtdb.ClientSessionBody) error {
 		}
 	}
 
-	// All justice transactions have a p2wkh output paying to the victim.
-	weightEstimate.AddP2WKHOutput()
+	// All justice transactions will either use segwit v0 (p2wkh + p2wsh)
+	// or segwit v1 (p2tr).
+	switch len(t.sweepPkScript) {
+	case input.P2WPKHSize:
+		weightEstimate.AddP2WKHOutput()
+
+	// P2WSH and P2TR are the same size: 34 bytes.
+	case input.P2WSHSize:
+		weightEstimate.AddP2WSHOutput()
+
+	default:
+		return fmt.Errorf("invalid sweep script length %d",
+			len(t.sweepPkScript))
+	}
 
 	// If the justice transaction has a reward output, add the output's
 	// contribution to the weight estimate.
 	if session.Policy.BlobType.Has(blob.FlagReward) {
-		weightEstimate.AddP2WKHOutput()
+		weightEstimate.AddP2TROutput()
```

Comment:
It's been a while since I looked at this code, but can't this be an older watchtower that uses p2wkh? It doesn't matter since you'd be overestimating (rather than under) the weight in this case but it wouldn't be accurate by a small amount of vbytes. Better to parse?

Reply:
Yeah my rationale here is that the new addresses are larger than the old, so worst case we over estimate it as you mentioned. 

Moved to just parse the address, since it seems to be the more bullet proof route here generally. I think before we used a garbage address here that wouldn't actually parse, but that's been fixed now in the tests.